### PR TITLE
Added two new endpoints and ability to handle query_string url keys/templates

### PR DESCRIPTION
**Pull request by drewbroadley**
Added two new endpoints:

- Report/BalanceSheetSummary ([docs](https://developer.myob.com/api/myob-business-api/v2/report/balancesheetsummary/))
- Report/ProfitAndLossSummary ([docs](https://developer.myob.com/api/myob-business-api/v2/report/profitandlosssummary/))

Improved handling for url keys/templates if there is a query string involved in endpoint.

### CODE REVIEW

**Comment by jarekwg** [comment]
Thanks for the contribution!
Are those queryparams mandatory?
I'm curious whether they belong in the endpoints definition, or whether they should be supported more how we support filters..

**Comment by drewbroadley** [comment]
For the Balance Sheet one, there's two of the three fields that are
mandatory:


   - The {AsOfDate} is the date of the end of the period. (required)
   - The {YearEndAdjust} states if you are includeding Year end
   Adjustments. This is a true or false value. (required)
   - The {ReportingBasis} will either be Cash or Accrual. (This filter is
   optional)

**Comment by jarekwg** [comment]
Lovely! Code looks great!

Queryparams in the endpoints file do make me a little uneasy, but given they're required, your solution is actually quite neat. We can always refactor in future to support queryparams better if more use cases were to pop up.

I also checked whether `requests` library is happy receiving queryparams in the `url` directly if there are others coming in via `params`, and it looks like it doesn't mind at all, and even composes them properly so that we don't get multiple `?` in there:

```
In [1]: requests.get('https://www.google.com/?q=hello', params={'r': 'goodbye'}).request.url
Out[1]: 'https://www.google.com/?q=hello&r=goodbye'
```

Could you please add corresponding tests to the endpoints file (following the existing format), and then I'll go ahead and merge and publish!

File: myob/endpoints.py [hunk 1]
```diff
@@ -72,6 +72,13 @@
             (GET, 'JournalTransaction/', 'transaction journal'),
         ]
     },
+    'Report/': {
+        'name': 'report',
```
Comment (jarekwg) [inline myob/endpoints.py:76]:
```suggestion
        'name': 'reports',
```

**Comment by drewbroadley** [comment]
Sorry for the silence. Yep, can write some tests this week and update PR.

**Comment by jarekwg** [comment]
Hey @drewbroadley I'm going to close this due to inactivity. Feel free to reopen if you get a moment to finish it off.

**Pull request closed without merging**